### ISSUE TITLE
List of Known Bugs

### ISSUE THREAD

**Issue by eorodrig** [CMPUT301W15T13/TravelPlanner#199]
Please list all the known bugs here. That way we could tackle these before the app is due and the app crashes on the demo date.

**Comment by bkhunter** [comment]
A ton of tests fail with nullPointer exceptions because of the persistance changes. I don't know if this is an easy bug to fix, but I can get most of them to work again by commenting out a few lines related to persistence in the ClaimList class.

**Comment by eorodrig** [comment]
Do not comment out the persistence calls. All current persistence calls only deal with local persistence.

Tests will have to be modified to accommodate persistence.

**Comment by eorodrig** [comment]
I looked through the code and i think the null pointer exceptions come from UI related requirements for local Persistence.

I created a testing mode to by-pass persistence.
DataManager.setTestMode();  <-- call this in the test method to shut off persistence

**Comment by bkhunter** [comment]
Awesome!

**Comment by bkhunter** [comment]
The approver comment functionality seems to have disappeared. Maybe @jihwan1ua can fix?

**Comment by lolicatgirls** [comment]
Hm, I found a strange bug where expenses are being shared across different users
I'm looking into it now but this is how it's replicated

Each expense is being bound to an index on a claim 
IE
Create an expense on the first claim of a user and create another on the second claim
Switch to a user that has no claims at the moment
Create a claim - The expense from the previous user is there
Create a second claim - The expense from the previous user (claim 2) is there

Not sure if its a bug on persistence or a model bug yet

**Comment by jihwan1ua** [comment]
Hmm, I will check on approver comment.
It should be there with all functionality

**Comment by eorodrig** [comment]
I think the expense bug is caused by a non persistance issue. The
persistance code does not modify claim ids or expense ids.

I think the bug might be in the activity or fragment. Possibly the model
for expense list.

**Comment by eorodrig** [comment]
Ill  check out if its a persistance issue later

**Comment by lolicatgirls** [comment]
Yeah for sure, I've been working on documentation and refactoring for the time being but I'll definitely take another look at it. Certainly a bit of a strange bug

**Comment by eorodrig** [comment]
Took a quick look at the expense issue (turned off persistence) and the issue is still there. Will try to track it down

**Comment by bkhunter** [comment]
Removed from expenseItemList: Hopefully not needed

	/**
	 * This method is to be implemented for project 5
	 */
	public void notifyListeners() {
		/**
		 *  TODO Auto-generated method stub	
		 */
	}

**Comment by bkhunter** [comment]
also commented out code in delete method of same class removed:

	//save the claim
		//	Claim claim = ClaimListSingleton.getClaimByID(claimID);
			//DataManager.saveClaim(claim);

**Comment by eorodrig** [comment]
Looked into expense bug and fixed some of it so far but it is quite embedded into the app.

Essentially, the bug arises on the way the app is coded.
ClaimListSingleton stores all the claims (from every user)
The fragments then deal with indices from the ClaimListSingleton instead of using the UUIID strings
This results in things being pointed towards the wrong claims in the list.

I made an attempt to fix some of it, but most of the code in expenses is based on claimIndex and expenseIndex so this will take a lot of time to refactor and correct.

What i fixed so far: 
I changed the code to rely on UUID when loading claims 
^-- this solves the problem James talked about.. 
     ^--- broke when saving claims due to index problems
I changed the code so that expenses are saved (on creation at least) to the claim with the same claimUUID
^--- solved the issue from above
    ^---Other Longclick/short click methods still fail <-- get the wrong expense.

//Worked on Editing as that is close to creating an expense
^-- kind of works, as in, the edited claim shows up in the current expense list.
     ^-- broken as the expense data viewed belongs to another claim.

//TODO: 
Will need to get someone who knows the layout of the expense activites to help me solve this otherwise this will take a long time to fix.

**Comment by bkhunter** [comment]
That's a huge pain...Well @lolicatgirls knows the expenseActivities layout best but he may be asleep. I can try to help if needed

**Comment by jihwan1ua** [comment]
There seems to be a problem upon returning a claim, it seems to work off-line, can't guess what is problem

**Comment by lolicatgirls** [comment]
oh, I see I see I see. For some reason I was assuming adding to the claimlist singleton by index worked due to the nature of the mode/datastructure(?) and with this assumption when we load this consistency should have been kept

so I was pretty stumped as far as what was causing the bug
I'll look into trying to fix this

**Comment by bkhunter** [comment]
Returning a claim is actually okay. However the app is crashing on a made expenseItem click. I hope this is possibly due to fixing other bugs and can be an easy fix.

**Comment by lolicatgirls** [comment]
I made the changes to the expense related fragments, However I'm having issues getting the expense inside of the photo activity. 

I'm following the same format as the rest of the expenses but everything will die after coming back after taking the image

Currently it is faked using the old code (claimIndex) 
and the code that is left over from playing around with it is commented out currently

**Comment by lolicatgirls** [comment]
oh god what have we done

**Comment by eorodrig** [comment]
Um.. is there a reason why the entire app has errors?

**Comment by mfritze** [comment]
You need to add google suport library that alex described, [here](https://github.com/CMPUT301W15T13/TravelPlanner/issues/201)

**Comment by lolicatgirls** [comment]
I think some of the users have bad data in the elastice search 
This is probably caused by weird saving of bitmaps to to the server
it should still work with new users

**Comment by lolicatgirls** [comment]
and by work I mean it doesn't explode with tons of errors
